x-pack/plugins/security/server/lib/authentication/authenticator.ts

```diff
@@ -195,7 +198,7 @@ class Authenticator {
    * Deauthenticates current request.
    * @param request Request instance.
    */
-  async deauthenticate(request: Legacy.Request) {
+  async deauthenticate(request: RequestWithLoginAttempt) {
```

Comment:
Yeah, I'm on the fence here. There were two reasons to use `RequestWithLoginAttempt` everywhere:

* When security is enabled request always includes `loginAttempt`, so technically it's more correct to have `RequestWithLoginAttempt` here
* We don't need to import any additional type from `kibana` (`import { Legacy.Request } from 'kibana'`) if we only use `RequestWithLoginAttempt`.

But none of these reasons are strong enough and since it confused you it will confuse others too, I'll switch back to `Legacy.Request` in `deauthenticate` just to not give a false impression that we may use `loginAttempt` where we shouldn't.